acl/acl.go

```diff
@@ -44,6 +47,10 @@ type ACL struct {
 	// namespaces maps a namespace to a capabilitySet
 	namespaces *iradix.Tree
 
+	// wildcardNamespaces maps a glob pattern of a namespace to a capabilitySet
+	// We use an iradix for the purposes of ordered iteration.
+	wildcardNamespaces *iradix.Tree
```

Comment:
Can we add documentation changes to this PR as well: 
https://github.com/hashicorp/nomad/blob/master/website/source/guides/security/acl.html.markdown

Reply:
Done - I'm not _too_ happy with my draft on those tho, so would love a second look from you / @Omar-Khawaja.

Reply:
@dantoml lgtm. I would probably change up the wording very slightly.

Namespace definitions may also include globs which can be used to provide access to a category of namespaces while allowing more specific control over privileges where required. For example, the below policy allows read access to most production namespaces but allows write access to the "production-api" namespace while denying access to the "production-ppi" namespace.